autorest/azure/token.go

```diff
@@ -168,12 +191,24 @@ func NewServicePrincipalTokenWithSecret(id string, tenantID string, resource str
 		refreshWithin: defaultRefresh,
 		sender:        &http.Client{},
 	}
-	return spt, nil
+	return spt
+}
+
+// NewServicePrincipalTokenFromManualToken creates a ServicePrincipalToken using the supplied token
+func NewServicePrincipalTokenFromManualToken(id string, tenantID string, resource string, token Token) *ServicePrincipalToken {
```

Comment:
hmm not sure why you would need new constructors? can't we just extend them with an extra nillable `refreshFunc` argument